server/opts.go

```diff
@@ -893,3 +916,260 @@ func processOptions(opts *Options) {
+				if opts.RoutesStr == "" {
+					// Set routes array to nil since routes string is empty
+					opts.Routes = nil
+					return
+				}
+				routeUrls := RoutesFromStr(opts.RoutesStr)
+				opts.Routes = routeUrls
+			}
+		}
+	})
+	if flagErr != nil {
+		return nil, flagErr
+	}
+
+	// This will be true if some of the `-tls` params have been set and
+	// `-tls=false` has not been set.
+	if tlsOverride {
+		if err := overrideTLS(opts); err != nil {
+			return nil, err
+		}
+	}
+
+	// If we don't have cluster defined in the configuration
+	// file and no cluster listen string override, but we do
+	// have a routes override, we need to report misconfiguration.
+	if opts.RoutesStr != "" && opts.Cluster.ListenStr == "" && opts.Cluster.Host == "" && opts.Cluster.Port == 0 {
+		return nil, errors.New("solicited routes require cluster capabilities, e.g. --cluster")
+	}
+
+	return opts, nil
+}
+
+// overrideTLS is called when at least "-tls=true" has been set.
+func overrideTLS(opts *Options) error {
+	if opts.TLSCert == "" {
+		return errors.New("TLS Server certificate must be present and valid")
+	}
+	if opts.TLSKey == "" {
+		return errors.New("TLS Server private key must be present and valid")
+	}
+
+	tc := TLSConfigOpts{}
+	tc.CertFile = opts.TLSCert
+	tc.KeyFile = opts.TLSKey
+	tc.CaFile = opts.TLSCaCert
+	tc.Verify = opts.TLSVerify
+
+	var err error
+	opts.TLSConfig, err = GenTLSConfig(&tc)
+	return err
+}
+
+// overrideCluster updates Options.Cluster if that flag "cluster" (or "cluster_listen")
+// has explicitly be set in the command line. If it is set to empty string, it will
+// clear the Cluster options.
+func overrideCluster(opts *Options) error {
+	if opts.Cluster.ListenStr == "" {
+		// This one is enough to disable clustering.
+		opts.Cluster.Port = 0
+		return nil
+	}
+	clusterURL, err := url.Parse(opts.Cluster.ListenStr)
+	if err != nil {
+		return err
+	}
+	h, p, err := net.SplitHostPort(clusterURL.Host)
+	if err != nil {
+		return err
+	}
+	opts.Cluster.Host = h
+	_, err = fmt.Sscan(p, &opts.Cluster.Port)
+	if err != nil {
+		return err
+	}
+
+	if clusterURL.User != nil {
+		pass, hasPassword := clusterURL.User.Password()
+		if !hasPassword {
+			return errors.New("expected cluster password to be set")
+		}
+		opts.Cluster.Password = pass
+
+		user := clusterURL.User.Username()
+		opts.Cluster.Username = user
+	} else {
+		// Since we override from flag and there is no user/pwd, make
+		// sure we clear what we may have gotten from config file.
+		opts.Cluster.Username = ""
+		opts.Cluster.Password = ""
+	}
+	return nil
+}
+
+func processSignal(fs *flag.FlagSet, signal string) error {
+	var (
+		pid           string
+		commandAndPid = strings.Split(signal, "=")
+	)
+	if l := len(commandAndPid); l == 2 {
+		pid = commandAndPid[1]
+	} else if l > 2 {
+		return fmt.Errorf("invalid signal parameters: %v", commandAndPid[2:])
+	}
+	if err := ProcessSignal(Command(commandAndPid[0]), pid); err != nil {
```

Comment:
I think the only reason `ProcessSignal` was exported was so it could be called from main.go. We could unexport it since it's not intended to be part of the public API, unless we consider that a breaking change.

Reply:
I don't think anybody would have use that, but it was indeed an exported function that is present in a public release (1.0.0+). Let me think about it.

Reply:
Also, I was passing the flagSet to this function because at one point during dev I needed it, which I don't right now. So will fix that for sure.